src/axom/core/tests/core_array_for_all.hpp

```diff
@@ -1204,6 +1204,15 @@ AXOM_TYPED_TEST(core_array_for_all, device_insert)
     typename TestFixture::template DynamicTArray<DynamicArray>;
 
   int kernelAllocID = axom::execution_space<ExecSpace>::allocatorID();
+#if defined(AXOM_USE_GPU) && defined(AXOM_USE_UMPIRE)
+  // Use unified memory for frequent movement between device operations
+  // and value checking on host
+  if(axom::execution_space<ExecSpace>::onDevice())
```

Comment:
Would it be helpful to remove the exec space check to make sure everything keep working for the "degenerate case" where data is only used on the host?